### PR TITLE
Upgrade node to version 12

### PR DESCRIPTION
**Pull request by garrettqmartin8**
# Description

This PR upgrades Node to version 12 so that we can keep upgrading react-uswds.

## Changelog or Releases

- [node 12.X](https://nodejs.org/en/about/releases/)
- [nodenv](https://github.com/nodenv/nodenv/releases) (added to `scripts/brew-upgrade`)

## Questions

What's the best way to release this while ensuring that master builds don't fail in the mymove repo? My rough plan is:

- Submit a PR on the mymove repo for my branch that uses the new circleci-docker image
- Once that's approved and merged, I'll post an announcement in #prac-engineering for everyone to merge master into their branches so that their builds target the new image
- Merge this PR

### CODE REVIEW

File: scripts/brew-upgrade [hunk 1]
```diff
@@ -12,7 +12,7 @@ brew upgrade \
     circleci \
     go \
     go-bindata \
-    node@10 \
+    nodenv \
```
Comment (chrisgilmerproj) [inline scripts/brew-upgrade:15]: How about instead we do `node@12`? Right now `nodenv` doesn't play nicely with this script.

**Comment by chrisgilmerproj** [comment]
> ## Questions
> 
> What's the best way to release this while ensuring that master builds don't fail in the mymove repo? My rough plan is:
> 
>     * Submit a PR on the mymove repo for my branch that uses the new circleci-docker image

Yep, this is the way. Make a PR and test it out before merging. You can also use docker-compose to pull down the built images from AWS and test them to see that the app works.

>     * Once that's approved and merged, I'll post an announcement in #prac-engineering for everyone to merge master into their branches so that their builds target the new image

Sounds good.

>     * Merge this PR

You can technically merge this whenever. Because on the transcom/mymove repo we use hash tags to identify what is used. So it won't automatically get used.

File: scripts/brew-upgrade (hunk 1 above)
Comment (garrettqmartin8) [reply scripts/brew-upgrade:15]: So part of the work I'm doing with the node upgrade is to recommend devs use nodenv to manage their node versions. I'll be adding an ADR along with the forthcoming PR on the mymove repo.

What did you see as far as nodenv not playing nicely with this script?

**Comment by garrettqmartin8** [comment]
> You can technically merge this whenever. Because on the transcom/mymove repo we use hash tags to identify what is used. So it won't automatically get used.

That makes, sense. Thanks for clarifying.

In my attempt to try to get everything to play nicely together, I targeted Node 12.16.3 specifically instead of 12.x because nodenv's latest supported 12.x version is 12.16.3, while debians package source mirror has 12.17. Alas, doing that here has [broken the build](https://circleci.com/gh/transcom/circleci-docker/107). @chrisgilmerproj any advice on how to target a specific version, or am I taking the wrong approach entirely?

File: scripts/brew-upgrade (hunk 1 above)
Comment (chrisgilmerproj) [reply scripts/brew-upgrade:15]: I suggest you try out the script and find out what steps folks need to make it work. Especially the part that sets the version, as in, can the script make sure it switches in advance of running.

**Comment by garrettqmartin8** [comment]
Nevermind, I think I was just requiring too-specific of a Node version.

File: scripts/brew-upgrade (hunk 1 above)
Comment (chrisgilmerproj) [reply scripts/brew-upgrade:15]: Additionally, if we move to nodenv then I think we have more to do here for upgrading: https://github.com/nodenv/nodenv#upgrading-with-homebrew

Comment (garrettqmartin8) [reply scripts/brew-upgrade:15]: Added `node-build` to the above script.

> can the script make sure it switches in advance of running

Not sure exactly what you mean. When people run `brew install nodenv` for the first time, they'll need to run `nodenv local 12.16.3`. The `check-node-version` script in the mymove repo will account for this. Are you saying that the local node version should be set automatically somehow?

Comment (chrisgilmerproj) [reply scripts/brew-upgrade:15]: This repo and script are supposed to be independent of the mymove repo. So if we can modify this script to run `nodenv local 12.16.3` that would be ideal.

Comment (garrettqmartin8) [reply scripts/brew-upgrade:15]: @chrisgilmerproj gotcha - so if we run the script here and it automatically runs `nodenv local`, that will only affect `circleci-docker` local Node version. We'd need to run `nodenv global` from the script in this repo if we wanted to affect the Node version in the mymove repo. Would we want to be setting the global Node version here? If so, I can add that to the script.

Comment (chrisgilmerproj) [reply scripts/brew-upgrade:15]: Since these repos are effectively split I only need it to affect this repo. We don't want to set the global version.

Comment (garrettqmartin8) [reply scripts/brew-upgrade:15]: I see what you mean now, thanks for the clarification. Will adjust the script.

**Comment by garrettqmartin8** [comment]
@chrisgilmerproj it looks like a curl issue - I was going to re-run the build but it's not even letting me run the build. Does that have anything to do with [this](https://ustcdp3.slack.com/archives/CP6PTUPQF/p1591292526088100)?

**Comment by chrisgilmerproj** [comment]
It ought to be running as this is a PR. Maybe you can work with @gidjin to figure out why this browser url is having issues?

**Comment by garrettqmartin8** [comment]
@chrisgilmerproj looks like the build is passing now - the old turn it off and on again trick worked :)

**Pull request merged**